src/WebJobs.Script/Workers/SharedMemoryDataTransfer/MemoryMappedFileAccessorLinux.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.SharedMemoryDataTransfer
+{
+    public class MemoryMappedFileAccessorLinux : MemoryMappedFileAccessor
+    {
+        public MemoryMappedFileAccessorLinux(ILogger<MemoryMappedFileAccessor> logger) : base(logger)
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
```

Comment:
nit: seems like this platform check logic can be factored into a private ValidatePlatform(OsPlatform) protected helper in the base class, since both implementations duplicate the logic.